smartlearner/views.py

```diff
@@ -51,3 +51,39 @@ def mean(self):
     @property
     def stderror(self):
         return ItemGetter(self, attribute=1)
+
+
+class ReconstructionError(View):
+    def __init__(self, predict_fct, dataset):
+        super().__init__()
+
+        batch_size = 1024  # Internal buffer
+        self.nb_batches = int(np.ceil(len(dataset) / batch_size))
+
+        input = dataset.symb_inputs
+        target = dataset.symb_targets
+        reconstruction_errors = T.sqr(predict_fct(input) - target).mean()
```

Comment:
All right, I got what you did. I thought this method was computing the 
*stderr* batch-wise, which was weird, but it's example-wise, which makes 
a lot more sense.

I'll amend my previous commit and change the logic of the code for the 
reconstruction error.

On 19/08/15 05:29 PM, Marc-Alexandre Côté wrote:
>
> In smartlearner/views.py 
> <https://github.com/SMART-Lab/smartlearner/pull/43#discussion_r37470705>:
>
> > @@ -51,3 +51,39 @@ def mean(self):
> >      @property
> >      def stderror(self):
> >          return ItemGetter(self, attribute=1)
> > +
> > +
> > +class ReconstructionError(View):
> > +    def __init__(self, predict_fct, dataset):
> > +        super().__init__()
> > +
> > +        batch_size = 1024  # Internal buffer
> > +        self.nb_batches = int(np.ceil(len(dataset) / batch_size))
> > +
> > +        input = dataset.symb_inputs
> > +        target = dataset.symb_targets
> > +        reconstruction_errors = T.sqr(predict_fct(input) - target).mean()
>
> Of course, if you don't care about the stderr, you don't have to keep 
> the reconstruction error of every examples. You can just take the mean 
> of the average reconstruction error of every batches.
>
> However, if you do want the stderr of your model predictions on the 
> dataset, you'd need to keep the reconstruction error of every examples 
> then compute the mean and the stderr.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/SMART-Lab/smartlearner/pull/43/files#r37470705>.Web Bug 
> from 
> https://github.com/notifications/beacon/AEJelAh-594WUn41kNHNPlSnewBKcSTkks5opOyngaJpZM4FuntY.gif
>